mpcpy/exodata.py

```diff
@@ -1332,7 +1337,101 @@ def _collect_data(self, start_time, final_time):
         # Process weather data
         if self.process_variables is not None:
             self._process_weather_data();   
-                                             
+
+class WeatherFromNOAA(_Weather, utility._DAQ):
+    '''Collects weather data from NOAA.
+    
+    It can either be historical or predicted weather data, depends on the start_time and final_time.
+    Based on the weather forecast function of pvlib version 6.0, https://pvlib-python.readthedocs.io/en/v0.6.0/
+
+    Parameters
+    ----------
+    geography: [numeric, numeric]
+        List of [Latitude, Longitude] in degrees.
+    weaForeModel: Weather forecast model, str,
+        GFS: Global Forecast System model, available for the entire globe and for 7 days ahead, support historical data, updated every 6 hours, 
+             time resolution: 3 hours, geographical resolutions: 0.25 and 0.5 deg 
+        HRRR: High Resolution Rapid Refresh model, available US and for ~15 hours ahead, DOES NOT support historical data, updated every hour, 
+             time resolution: 1 hour, geographical resolutions: 3 km
+        RAP: Rapid Refresh model, available US and for 2 days ahead (tested 18 hours), support historical data, updated every hour, 
+             time resolution: 1 hour, geographical resolutions: 20, 40 km
+        NAM: North American Mesoscale model, available for the whole North America and for 4 days ahead (tested 3 days), support historical data, 
+             updated every 6 hours, time resolution: 1 hour, geographical resolutions: 20 km
+    tz_name: timezone name, str
+        Default is 'from geography', then the timezone would be inferred automatically from the input geography
+        Users could also specify the timezone manually, example 'America/Chicago', 'UTC'
+
+    Attributes
+    ----------
+    data : dictionary
+        {"Weather Variable Name" : mpcpy.Variables.Timeseries}.
+    lat : mpcpy.variables.Static
+        Latitude in degrees.
+    lon : mpcpy.variables.Static
+        Longitude in degrees.
+    tz_name : string
+        Timezone name.  
+
+    '''
+    
+    def __init__(self, geography, weaForeModel, tz_name='from_geography', **kwargs):
+        '''Constructor of DataFrame weather exodata object.
+        
+        '''
+        
+        self.name = 'weather_from_noaa';
+        self.data = {};   
+        # Dictionary of format {'dfHeader' : ('weaVarName', mpcpyUnit)}
+        self.variable_map = {'temp_air'     : ('weaTDryBul', units.degC),
+                             'wind_speed'   : ('weaWinSpe', units.m_s),
+                             'ghi'          : ('weaHGloHor', units.W_m2),
+                             'dni'          : ('weaHDirNor', units.W_m2),
+                             'dhi'          : ('weaHDifHor', units.W_m2),
+                             'total_clouds' : ('weaNTot', units.percent),
+                             };
+        self.geography = geography
```

Comment:
See comment below about using lines 1319-1321 of ``exodata.py`` for guidance on using of ``geography`` to create ``self.lat`` and ``self.lon``.  There are MPCPy methods that exist already for this, and I want to try to use a single instance of baseclass code to do the same function in classes that need this functionality.